packages/angular_devkit/build_angular/src/builders/browser-esbuild/index.ts

```diff
@@ -0,0 +1,324 @@
+      });
+      await fs.writeFile(path.join(outputPath, sheetPath), sheetResult.contents);
+      if (sheetResult.map) {
+        outputFiles.push({
+          path: sheetPath + '.map',
+          text: sheetResult.map,
+          contents: Buffer.from(sheetResult.map, 'utf-8'),
+        });
+      }
+      if (!noInjectNames.includes(name)) {
+        initialFiles.push({
+          file: sheetPath,
+          name,
+          extension: '.css',
+        });
+      }
+      outputFiles.push(...sheetResult.resourceFiles);
+    }
+  }
+
+  // Generate index HTML file
+  if (options.index) {
+    const entrypoints = generateEntryPoints({
+      scripts: options.scripts ?? [],
+      styles: options.styles ?? [],
+    });
+
+    // TODO: Setup this to read from in-memory output files
+    const indexHtmlGenerator = new IndexHtmlGenerator({
+      indexPath: path.join(context.workspaceRoot, getIndexInputFile(options.index)),
+      entrypoints,
+      sri: options.subresourceIntegrity,
+      optimization: optimizationOptions,
+      crossOrigin: options.crossOrigin,
+    });
+
+    const { content, warnings, errors } = await indexHtmlGenerator.process({
+      baseHref: options.baseHref,
+      lang: undefined,
+      outputPath,
+      files: initialFiles,
+    });
+
+    for (const error of errors) {
+      context.logger.error(error);
+    }
+    for (const warning of warnings) {
+      context.logger.warn(warning);
+    }
+
+    outputFiles.push({
+      path: getIndexOutputFile(options.index),
+      text: content,
+      contents: Buffer.from(content, 'utf-8'),
+    });
+  }
+
+  // Copy assets
+  if (assets) {
+    await copyAssets(assets, [outputPath], workspaceRoot);
+  }
+
+  // Write output files
+  await Promise.all(
+    outputFiles.map((file) => fs.writeFile(path.join(outputPath, file.path), file.contents)),
+  );
+
+  context.logger.info(`Complete. [${(Date.now() - startTime) / 1000} seconds]`);
+
+  return { success: true };
+}
+
+/**
+ * Determines if an unknown value is an esbuild BuildFailure error object thrown by esbuild.
+ * @param value A potential esbuild BuildFailure error object.
+ * @returns `true` if the object is determined to be a BuildFailure object; otherwise, `false`.
+ */
+function isEsBuildFailure(value: unknown): value is BuildFailure {
+  return !!value && typeof value === 'object' && 'errors' in value && 'warnings' in value;
+}
+
+async function bundleCode(
+  workspaceRoot: string,
+  entryPoints: Record<string, string>,
+  outputNames: { bundles: string; media: string },
+  options: BrowserBuilderOptions,
+  optimizationOptions: NormalizedOptimizationOptions,
+  sourcemapOptions: SourceMapClass,
+  tsconfig: string,
+) {
+  let result;
+  try {
+    result = await build({
+      absWorkingDir: workspaceRoot,
+      bundle: true,
+      format: 'esm',
+      entryPoints,
+      entryNames: outputNames.bundles,
+      assetNames: outputNames.media,
+      target: 'es2020',
```

Comment:
shouldn't this be configurable based on the tsconfig target?

Reply:
Output target is now adjusted in the esbuild plugin setup.